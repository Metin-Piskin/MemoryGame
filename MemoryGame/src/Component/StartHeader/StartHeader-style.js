import { StyleSheet } from "react-native";

export default StyleSheet.create({
    innercontainer: {
        alignItems: 'center',
        borderRadius: 20
    },
    LinearGradientcontainer: {
        padding: 20,
        borderRadius: 120
    },
    title: {
        fontSize: 32,
        color: '#fff',
        fontFamily: 'PressStart2P-Regular'
    },
});