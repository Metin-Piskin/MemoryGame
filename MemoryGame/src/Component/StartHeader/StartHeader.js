import React from 'react';
import { View, Text } from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';

import styles from './StartHeader-style';

const StartHeader = () => {
    return (
        <View
            style={styles.innercontainer}
        >
            <LinearGradient
                style={styles.LinearGradientcontainer}
                colors={[
                    '#1e293b',
                    '#1e293b',
                    '#0f172a',
                    '#0f172a',
                ]}
            >
                <MaterialCommunityIcons
                    name='react'
                    color={'#fff'}
                    size={200}
                />
            </LinearGradient>
            <Text style={styles.title}>
                Memory Game
            </Text>
        </View>
    )
}

export default StartHeader;