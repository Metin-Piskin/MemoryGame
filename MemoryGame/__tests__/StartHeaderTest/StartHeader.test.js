import React from 'react';
import { render } from '@testing-library/react-native';

import StartHeader from '../../src/Component/StartHeader';
import StartHeaderStyles from '../../src/Component/StartHeader/StartHeader-style';

jest.mock('react-native-linear-gradient', () => 'LinearGradient')
jest.mock('react-native-vector-icons/MaterialCommunityIcons', () => 'MaterialCommunityIcons')

test('renders correctly', () => {
    const comp = render(<StartHeader />);
    expect(comp).toMatchSnapshot();
});


test('should render default theme style', () => {
    const comp = render(<StartHeader />);
    const StartHeaderContainerStyle = comp.getByTestId('start-header-container').props.style;

    expect(StartHeaderContainerStyle).toMatchObject(StartHeaderStyles.innercontainer);
});

test('should render default theme style', () => {
    const comp = render(<StartHeader />);
    const StartHeaderLinearGradientcontainerStyle = comp.getByTestId('start-header-LinearGradientcontainer').props.style;

    expect(StartHeaderLinearGradientcontainerStyle).toMatchObject(StartHeaderStyles.LinearGradientcontainer);
});

test('should render default theme style', () => {
    const comp = render(<StartHeader />);
    const StartHeaderTitleStyle = comp.getByTestId('start-header-title').props.style;

    expect(StartHeaderTitleStyle).toMatchObject(StartHeaderStyles.title);
});
