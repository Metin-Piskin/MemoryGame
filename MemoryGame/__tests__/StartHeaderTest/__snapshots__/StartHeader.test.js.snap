// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "borderRadius": 20,
    }
  }
  testID="start-header-container"
>
  <LinearGradient
    colors={
      [
        "#1e293b",
        "#1e293b",
        "#0f172a",
        "#0f172a",
      ]
    }
    style={
      {
        "borderRadius": 120,
        "padding": 20,
      }
    }
    testID="start-header-LinearGradientcontainer"
  >
    <MaterialCommunityIcons
      color="#fff"
      name="react"
      size={200}
    />
  </LinearGradient>
  <Text
    style={
      {
        "color": "#fff",
        "fontFamily": "PressStart2P-Regular",
        "fontSize": 32,
      }
    }
    testID="start-header-title"
  >
    Memory Game
  </Text>
</View>
`;
